network/dns.go

```diff
@@ -29,33 +29,38 @@ const (
 	openDNSRequestLimit = 3 * time.Second
 )
 
-func getDNSRequestCacheKey(pid int, fqdn string) string {
-	return strconv.Itoa(pid) + "/" + fqdn
+func getDNSRequestCacheKey(pid int, fqdn string, qType uint16) string {
+	return strconv.Itoa(pid) + "/" + fqdn + dns.Type(qType).String()
 }
 
 func removeOpenDNSRequest(pid int, fqdn string) {
 	openDNSRequestsLock.Lock()
 	defer openDNSRequestsLock.Unlock()
 
-	key := getDNSRequestCacheKey(pid, fqdn)
-	_, ok := openDNSRequests[key]
-	if ok {
-		delete(openDNSRequests, key)
-		return
-	}
+	// Delete PID-specific requests.
+	delete(openDNSRequests, getDNSRequestCacheKey(pid, fqdn, dns.TypeA))
+	delete(openDNSRequests, getDNSRequestCacheKey(pid, fqdn, dns.TypeAAAA))
```

Comment:
Shouldn't we only remove the DNS request for which we recieved the response? We now delete all `A` and `AAAA` requests for the domain which might cause the next answer  received to not have a open-DNS request available